frontend/packages/core/src/dialog.tsx

```diff
@@ -64,19 +64,29 @@ export interface DialogProps extends Pick<MuiDialogProps, "open"> {
     | React.ReactElement<DialogContentProps>[]
     | React.ReactElement<DialogActionsProps>
     | React.ReactElement<DialogActionsProps>[];
+  onClose?: () => void;
 }
 
-const Dialog = ({ title, children, open }: DialogProps) => {
-  const [isOpen, setIsOpen] = React.useState(open);
-  const onClose = () => {
-    setIsOpen(prevOpenState => !prevOpenState);
+const Dialog = ({ title, children, open: isOpenFromProps, onClose }: DialogProps) => {
+  const [isOpenFromState, setIsOpen] = React.useState(isOpenFromProps);
+  const onCloseHandlerExists = typeof onClose === "function";
```

Comment:
I dont know my typescript very well but is there a const for datatypes we could use over the string?

Reply:
hmm... i guess I did define it in the `DialogProps` interface already so the typeof comparison doesn't have to be done... would `const onCloseHandlerExists = onClose ? true : false;` be better?

Reply:
I will defer to you but that also looks good!

Reply:
okay pushed out a new commit!